### PR TITLE
feat: write rolling functions as private expressions. 

### PR DESCRIPTION
**Pull request by uchiiii**
To fix https://github.com/pola-rs/polars/issues/10101

### CODE REVIEW

File: crates/polars-plan/src/dsl/function_expr/rolling.rs [hunk 1]
```diff
@@ -1,5 +1,185 @@
...
+                window_size.hash(state);
+                bias.hash(state)
+            },
+            _ => {},
+        }
+    }
+}
+
+macro_rules! convert {
+    ($func:ident, $s:expr, $options:expr) => {{
```
Comment (ritchie46) [inline crates/polars-plan/src/dsl/function_expr/rolling.rs:72]: Can we use a generic function for this?

**Review by ritchie46 — CHANGES_REQUESTED** [review]
Nice! @uchiiii. This looks great already.

I have one comment about the macro. I don't like code in macros and find them hard to maintain and reason about. Can you rewrite that part to use generics.

File: crates/polars-plan/src/dsl/function_expr/rolling.rs (hunk 1 above)
Comment (uchiiii) [reply crates/polars-plan/src/dsl/function_expr/rolling.rs]: I tried not to use macro. If there is something wired or wrong, let me know!

https://github.com/pola-rs/polars/pull/12379/commits/ece67b12db5d63a20759064a15037efff4322419

**Review by ritchie46 — APPROVED** [review]

File: crates/polars-plan/src/dsl/function_expr/rolling.rs (hunk 1 above)
Comment (ritchie46) [reply crates/polars-plan/src/dsl/function_expr/rolling.rs]: Thank you. Great PR. :+1:

**Pull request merged**